st2client/tests/unit/test_command_actionrun.py

```diff
@@ -55,3 +56,81 @@ def test_get_params_types(self):
         self.assertTrue('stuff' not in imm, '"stuff" param should be in immutable set.')
         self.assertEqual(runner.runner_parameters, orig_runner_params, 'Runner params modified.')
         self.assertEqual(action.parameters, orig_action_params, 'Action params modified.')
+
+    def test_get_params_from_args(self):
+        runner = RunnerType()
+        runner.runner_parameters = {}
+
+        action = Action()
+        action.ref = 'test.action'
+        action.parameters = {
+            'param_string': {'type': 'string'},
+            'param_integer': {'type': 'integer'},
+            'param_number': {'type': 'number'},
+            'param_object': {'type': 'object'},
+            'param_boolean': {'type': 'boolean'},
+            'param_array': {'type': 'array'},
+            'param_array_of_dicts': {'type': 'array'},
+        }
+
+        subparser = mock.Mock()
+        command = ActionRunCommand(action, self, subparser, name='test')
+
+        mockarg = mock.Mock()
+        mockarg.inherit_env = False
+        mockarg.parameters = [
+            'param_string=hoge',
+            'param_integer=123',
+            'param_number=1.23',
+            'param_object=hoge=1,fuga=2',
+            'param_boolean=False',
+            'param_array=foo,bar,baz',
+            'param_array_of_dicts=foo:1,bar:2'
```

Comment:
How does it work if you want to specify multiple objects for an array?

Would something like that work: `param_array_of_dicts=foo:1,bar:2,aram_array_of_dicts=foo:3,bar:4`.

If yes, let's please add a test case or it and also document it.